scripts/agave-build-lists.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+# Defines reusable lists of Agave binary names for use across scripts.
+
+# Source this file to access the arrays
+# Example:
+#   source "scripts/agave-build-lists.sh"
+#   printf '%s\n' "${AGAVE_BINS_DEV[@]}"
+
+
+# Groups with binary names to be built, based on their intended audience
+# Keep names in sync with build/install scripts that consume these lists.
+
+# shellcheck disable=SC2034
+AGAVE_BINS_DEV=(
+  cargo-build-sbf
+  cargo-test-sbf
+  solana-stake-accounts
```

Comment:
Can this go with `AGAVE_BINS_END_USER`? It doesn't make a huge difference either way, but it should belong with solana-tokens at the very least

Reply:
Reason why I put it here is so that it doesn't get included in the `net.sh` build.

I don't think it's a long build time so we can add it there if it's better suited.

Reply:
Would both of them being in `DEV` be more acceptable?

Reply:
Oh gotcha, maybe we can create a separate "AGAVE_BINS_TO_DEPRECATE" category just for those two, and not include them in net.sh builds :smile:

Reply:
Sounds good. I created a separate group for deprecated bins that we probably should remove.